File: VB/FileManager_Example/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports FileManager_Example.Helpers

Namespace FileManager_Example.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!"
			Return View()
		End Function
		<ValidateInput(False)>
		Public Function FileManagerPartial() As ActionResult
			Return PartialView("_FileManagerPartial", HomeControllerFileManagerSettings.Model)
		End Function

		Public Function FileManagerPartialDownload() As FileStreamResult
			Return FileManagerExtension.DownloadFiles(FMSettingsHelper.CreateFileManagerDownloadSettings(), HomeControllerFileManagerSettings.Model)
		End Function
	End Class
	Public Class HomeControllerFileManagerSettings
		Public Const RootFolder As String = "~\"

		Public Shared ReadOnly Property Model() As String
			Get
				Return RootFolder
			End Get
		End Property
	End Class
End Namespace

File: CS/FileManager_Example/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using FileManager_Example.Helpers;

namespace FileManager_Example.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Message = "Welcome to DevExpress Extensions for ASP.NET MVC!";
            return View();
        }
        [ValidateInput(false)]
        public ActionResult FileManagerPartial() {
            return PartialView("_FileManagerPartial", HomeControllerFileManagerSettings.Model);
        }

        public FileStreamResult FileManagerPartialDownload() {
            return FileManagerExtension.DownloadFiles(FMSettingsHelper.CreateFileManagerDownloadSettings(), HomeControllerFileManagerSettings.Model);
        }
    }
    public class HomeControllerFileManagerSettings {
        public const string RootFolder = @"~\";

        public static string Model { get { return RootFolder; } }
    }
}